packages/compiler/src/install/install.ts

```diff
@@ -0,0 +1,198 @@
+import { fork } from "child_process";
+import { mkdir, rename, rm } from "fs/promises";
+import { homedir } from "os";
+import type { CliCompilerHost } from "../core/cli/types.js";
+import { DiagnosticError } from "../core/diagnostic-error.js";
+import { createDiagnosticCollector } from "../core/index.js";
+import { createTracer } from "../core/logger/tracer.js";
+import { joinPaths } from "../core/path-utils.js";
+import { NoTarget, type Diagnostic, type Tracer } from "../core/types.js";
+import {
+  downloadAndExtractPackage,
+  fetchPackageManifest,
+} from "../package-manger/npm-registry-utils.js";
+import { mkTempDir } from "../utils/fs-utils.js";
+import { getPackageManagerConfig, type PackageManagerConfig } from "./config.js";
+import {
+  resolvePackageManagerSpec,
+  updatePackageManagerInPackageJson,
+  type ResolvedSpecResult,
+} from "./spec.js";
+
+interface SpawnError {
+  errno: number;
+  code: "ENOENT";
+  sysCall: string;
+  path: string;
+  spawnArgs: string[];
+}
+
+const tspDir = joinPaths(homedir(), ".tsp");
```

Comment:
The practice of using `~/.{thing}` is not really considered good practice anymore. I think it will be better to use

- Linux: `~/.local/share/typespec`
- Windows: `%APPDATA%\Local\typespec`
- macOS: `~/Library/Application Support/typespec`